proxy.php

```diff
@@ -56,8 +56,8 @@
 // the HTTP response headers captured via fopen or curl
 $httpResponseHeaders = [];
 
-// 1) PIWIK.JS PROXY: No _GET parameter, we serve the JS file
-if (empty($_GET) && empty($_POST)) {
+// 1) PIWIK.JS PROXY: No _GET parameter, we serve the JS file; or we serve a requested js file
+if ((empty($_GET) && empty($_POST)) || (isset($filerequest) && substr($filerequest, -3) !== false && substr($filerequest, -3) === '.js')) {
```

Comment:
This can be simplified to `isset($filerequest) && substr($filerequest, -3) === '.js'`